python/kserve/pyproject.toml

```diff
@@ -52,6 +52,9 @@ tabulate = "^0.9.0"
 pandas =  "^2.2.0"
 pydantic = [{ version = "^2.5.0", python = "3.12" }, { version = ">1.0,<3", python = ">=3.9,<3.12" }]
 pyyaml = "^6.0.0"
+torch = "~2.4.0"
```

Comment:
This would introduce unwanted torch dependency for all the other serving runtimes.

Reply:
I think I can remove it

Reply:
@sivanantha321 removed, on finishing mock tests for vLLM models.

Reply:
@sivanantha321 Hi, I am sorry to trouble you! I feel it would be a problem to write the mock tests for vLLM models, as it requires `get_tokenizer` from engine_client(async_engine), and give a `mock_get_tokenizer`, we need more objected exposed. Do you have some ideas on how to avoid this on mock tests for vLLM. @gavrissh When should I expect that we can use the vllm cpu to test the performance?

Reply:
@hustxiayang @gavrissh Does vLLM cpu build supports AMD cpus ? Because the last time I checked github runners are using AMD cpus. I am not able to run the vLLM cpu build in the CI. 

I think we didn't mock the tokenizer for the mock tests. Do we need to mock it now ?

Reply:
>Does vLLM cpu build supports AMD cpus 

I don't think it supports AMD CPUs 

Just looked vLLM OpenVINO build might have support, need to confirm - https://github.com/openvinotoolkit/openvino/issues/23043

Reply:
@sivanantha321 we need to mock it  for vLLM's `OpenAIServingChat` as it calls `get_tokenizer()` from engine_client/AsyncEngine. @gavrissh if vLLM-cpu test is not close to be ready, I may have to figure out to use mock tests. @sivanantha321 Another question is that, previously, Kserve implemented it's own servingengine using AsyncEngine, so it makes sense to write test codes on whether servingengine can AsyncEngine in a correct way. Now, we just using vLLM's existing engine, do we still to have these tests? Probably most tests are not necessary

Reply:
@hustxiayang, as I understand, the vLLM cpu build can't run on GitHub runners as they use AMD CPUs. Don't think it will help you with this scenario.

Reply:
@gavrissh thanks a lot for confirming it

Reply:
@hustxiayang I am going to try with OpenVINO build for e2e test. For now let's focus on the implementation.